### ISSUE TITLE
How to store these emoji's to SQLite database and retrieve them?

### ISSUE THREAD

**Issue by curiousmit** [rockerhieu/emojicon#12]
Problem storing these emoji's to SQLite database and retrieving them as these emoji's support utf8mb4 encoding which is not supported by SQLite, i suppose. Currently if i try to store and retrieve from database it gives a ?? for each emoji.

How to store and retrieve these to android SQLite database.

**Comment by gorio** [comment]
I have the same problem.

Does Anyone got any solution ?

I need to save message from EmojiconEditText to database and after get data and show in a EmojiconTextView.

Is it possible ?

**Comment by themitshah** [comment]
Yes indeed i have found one. The data type for the emoji to be stored on
server, needs to be kept as NVARCHAR or equivalent. For me, that solved the
problem of storing and retrieving emoji to and from db.

**Comment by gorio** [comment]
Have you used SQLite ?

**Comment by gorio** [comment]
How to get text from EmojiconEditText ?

EmojiconEditText etMessage;
etMessage.getText().toString()

**Comment by gorio** [comment]
@themitshah could you contact me goriofab@gmail.com ? Thank you

**Comment by gorio** [comment]
![screenshot_2015-09-04-15-31-52](https://cloud.githubusercontent.com/assets/3804913/9691685/62202a40-531a-11e5-8708-831c6f6e7d4c.png)

**Comment by seema7** [comment]
you may store the emoticons from edit text as 
StringEscapeUtils.escapeJava(edittext.getText().toString());  to get it in unicode format.

and decode it back to emoticon u can use 
textview.setText(StringEscapeUtils.unescapeJava(<unicode_emoticon>));

**Comment by Wumingla** [comment]
i through "rails new scaffold p name :string description:text" and i use the sqlite,when the description has emoji,   the sqlite can not store it ,how to deal it?thanks

**Comment by AljoSt** [comment]
I don't know if this is applicable to SQLite, but I just get the String from the EditText (`EmojiconEditText.getText().toString()`), and then de/encode the string as described [here](http://stackoverflow.com/questions/25074112/how-to-encode-and-decode-emoji-in-android/28273410#28273410)

**Comment by seema7** [comment]
You can save the emoticons into Sqlite db by saving the emoticon's emoji i.e use StringEscapeUtils.escapeJava(EmojiconEditText.getText().toString()); 
to get the uniocode of the emoji and save it to Sqlite.
To retrieve back the emoticon from that unicode use 
StringEscapeUtils.unescapeJava(<unicode_emoticon>); 
and set the value received from above to a textview which displays the required emoticon ..

**Comment by szuwest** [comment]
> Yes indeed i have found one. The data type for the emoji to be stored on server, needs to be kept as NVARCHAR or equivalent. For me, that solved the problem of storing and retrieving emoji to and from db. On 4 Sep 2015 21:39, "Eduardo Gorio" [notifications@github.com](mailto:notifications@github.com) wrote:
> 
> > I have the same problem.
> > Does Anyone got any solution ?
> > I need to save message from EmojiconEditText to database and after get
> > data and show in a EmojiconTextView.
> > Is it possible ?
> > —
> > Reply to this email directly or view it on GitHub
> > [#12 (comment)](https://github.com/rockerhieu/emojicon/issues/12#issuecomment-137778798).

can you post your 'create table sql'
